core/src/main/java/io/undertow/io/BlockingSenderImpl.java

```diff
@@ -292,11 +302,15 @@ private void invokeOnComplete(final IoCallback callback) {
             } else if (file != null) {
                 performTransfer(file, queuedCallback);
             }
-            inCall = true;
+            sendThread = null;
+            inCall = Thread.currentThread();
             try {
-                queuedCallback.onComplete(exchange, this);
+                callback.onComplete(exchange, this);
```

Comment:
@stuartwdouglas  Since we are on next != null and have already invoked callback, shouldn't t this line be kept as queuedCallback?

Reply:
yes, that is a copy/paste error